### PR TITLE
Add the shunt compensators modifications in the voltage init modification

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java [hunk 1]
```diff
@@ -185,6 +199,89 @@ private void applyVscConverterStationModification(Network network, Reporter subR
         });
     }
 
+    private void applyShuntCompensatorModification(Network network, Reporter subReporter) {
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java:202]: The algo is a bit hard to read and has some duplicate lines. I propose you below a shortest version, strictly identical.

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java [hunk 2]
```diff
@@ -185,6 +199,89 @@ private void applyVscConverterStationModification(Network network, Reporter subR
...
+                                reports.add(createReport("shuntCompensatorDisconnected", "Shunt compensator disconnected", Map.of(), TypedValue.INFO_SEVERITY, 1));
+                                shuntCompensator.setSectionCount(0);
+                                reports.add(ModificationUtils.getInstance().buildModificationReportWithIndentation(currentSectionCount, shuntCompensator.getSectionCount(), SECTION_COUNT, 1));
+                            } else if (m.getSectionCount() > 1) {
+                                reports.add(createReport(SHUNT_SECTION_COUNT_VALUE_IGNORED_KEY, SHUNT_SECTION_COUNT_VALUE_CANNOT_BE_APPLIED, Map.of(VALUE, m.getSectionCount()), TypedValue.WARN_SEVERITY, 1));
+                            }
+                        } else {
+                            reports.add(createReport("shuntCompensatorCurrentSectionCountValueIgnored", "Current section count value ${value} should be 0 or 1", Map.of(VALUE, currentSectionCount), TypedValue.WARN_SEVERITY, 1));
+                        }
+                    }
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java:242]:
```suggestion
                    if (m.getSectionCount() == null) {
                        reports.add(createReport("shuntCompensatorSectionCountUndefined", "Section count value is undefined", Map.of(), TypedValue.WARN_SEVERITY, 1));
                    } else if (m.getSectionCount() > 1) {
                        reports.add(createReport(SHUNT_SECTION_COUNT_VALUE_IGNORED_KEY, SHUNT_SECTION_COUNT_VALUE_CANNOT_BE_APPLIED, Map.of(VALUE, m.getSectionCount()), TypedValue.WARN_SEVERITY, 1));
                    } else if (currentSectionCount > 1) {
                        reports.add(createReport("shuntCompensatorCurrentSectionCountValueIgnored", "Current section count value ${value} should be 0 or 1", Map.of(VALUE, currentSectionCount), TypedValue.WARN_SEVERITY, 1));
                    } else {
                        if (m.getSectionCount() == 0) {
                            shuntCompensatorTerminal.disconnect();
                            reports.add(createReport("shuntCompensatorDisconnected", "Shunt compensator disconnected", Map.of(), TypedValue.INFO_SEVERITY, 1));
                        }
                        if (m.getSectionCount() != currentSectionCount) {
                            shuntCompensator.setSectionCount(m.getSectionCount());
                            reports.add(ModificationUtils.getInstance().buildModificationReportWithIndentation(currentSectionCount, m.getSectionCount(), SECTION_COUNT, 1));
                        }
                    }
```

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java [hunk 3]
```diff
@@ -185,6 +199,89 @@ private void applyVscConverterStationModification(Network network, Reporter subR
...
+                                    shuntCompensatorTerminal.connect();
+                                    reports.add(createReport("shuntCompensatorReconnected", "Shunt compensator reconnected", Map.of(), TypedValue.INFO_SEVERITY, 1));
+                                }
+                            } else if (m.getSectionCount() > 1) {
+                                reports.add(createReport(SHUNT_SECTION_COUNT_VALUE_IGNORED_KEY, SHUNT_SECTION_COUNT_VALUE_CANNOT_BE_APPLIED, Map.of(VALUE, m.getSectionCount()), TypedValue.WARN_SEVERITY, 1));
+                            }
+                        } else {
+                            reports.add(createReport("shuntCompensatorCurrentSectionCountValueIgnored", "Current section count value ${value} should be 0 or 1", Map.of(VALUE, currentSectionCount), TypedValue.WARN_SEVERITY, 1));
+                        }
+                    }
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java:273]:
```suggestion
                    if (m.getConnect() == null) {
                        reports.add(createReport("shuntCompensatorConnectUndefined", "Connect value is undefined", Map.of(), TypedValue.WARN_SEVERITY, 1));
                    } else if (m.getSectionCount() > 1) {
                        reports.add(createReport(SHUNT_SECTION_COUNT_VALUE_IGNORED_KEY, SHUNT_SECTION_COUNT_VALUE_CANNOT_BE_APPLIED, Map.of(VALUE, m.getSectionCount()), TypedValue.WARN_SEVERITY, 1));
                    } else if (currentSectionCount > 1) {
                        reports.add(createReport("shuntCompensatorCurrentSectionCountValueIgnored", "Current section count value ${value} should be 0 or 1", Map.of(VALUE, currentSectionCount), TypedValue.WARN_SEVERITY, 1));
                    } else {
                        if (Boolean.TRUE.equals(m.getConnect()) && m.getSectionCount() == 1) {
                            shuntCompensatorTerminal.connect();
                            reports.add(createReport("shuntCompensatorReconnected", "Shunt compensator reconnected", Map.of(), TypedValue.INFO_SEVERITY, 1));
                        }
                        if (m.getSectionCount() != currentSectionCount) {
                            shuntCompensator.setSectionCount(m.getSectionCount());
                            reports.add(ModificationUtils.getInstance().buildModificationReportWithIndentation(currentSectionCount, m.getSectionCount(), SECTION_COUNT, 1));
                        }
                    }
```

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: I propose a code simplification
tests: ok, but one remark (maybe a known issue):
 - in the logs tab, after modification build, we cannot see the MCS modifications in the right/tree part. We only see "Generator xxx" items.  EDIT: I meant in the "left/tree part" ;-)

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java]: Done

**Comment by FranckLecuyer** [comment]
> code: I propose a code simplification tests: ok, but one remark (maybe a known issue):
> 
>     * in the logs tab, after modification build, we cannot see the MCS modifications in the right/tree part. We only see "Generator xxx" items.

To have modifications on shunts, you must provide shunt compensators filter(s) in the settings : 
Voltage profile initialization --> Equipment selection --> Variable shunt compensators

Otherwise, no shunt will be modified ...

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Pull request merged**